skrub/datasets/_fetchers.py

```diff
@@ -0,0 +1,419 @@
+"""
```

Comment:
I'm not super enthousiastic about having a file called "_fetching.py" and one called "_fetcher.py". It's very unclear from the name of the files in which file should go what.

Reply:
What about `_fetching_backend.py` and `_fetching_functions.py`? And to further clarify, add docstrings to the file, like we do for others?